frontend/src/pages/projects/projectSharing/ProjectSharingTableRow.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import { ActionsColumn, Tbody, Td, Tr } from '@patternfly/react-table';
+import {
+  Button,
+  DropdownDirection,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Text,
+} from '@patternfly/react-core';
+import { CheckIcon, TimesIcon } from '@patternfly/react-icons';
+import { RoleBindingKind } from '~/k8sTypes';
+import { castProjectSharingRoleType, formatDate } from './utils';
+import { ProjectSharingRoleType } from './types';
+
+type ProjectSharingTableRowProps = {
+  obj: RoleBindingKind;
+  isEditing: boolean;
+  typeAhead: string[];
+  onCreateOrEditRoleBinding: (
+    name: string,
+    roleType: ProjectSharingRoleType,
+    rolebinding: RoleBindingKind,
+  ) => void;
+  onCancelRoleBindingCreation: () => void;
+  onEditRoleBinding: (rolebinding: RoleBindingKind) => void;
+  onDeleteRoleBinding: (rolebinding: RoleBindingKind) => void;
+};
+
+const ProjectSharingTableRow: React.FC<ProjectSharingTableRowProps> = ({
+  obj,
+  isEditing,
+  typeAhead,
+  onCreateOrEditRoleBinding,
+  onCancelRoleBindingCreation,
+  onEditRoleBinding,
+  onDeleteRoleBinding,
+}) => {
+  const [roleBindingName, setRoleBindingName] = React.useState(obj.subjects[0]?.name || '');
+  const [roleBindingRoleRef, setRoleBindingRoleRef] = React.useState<ProjectSharingRoleType>(
+    castProjectSharingRoleType(obj.roleRef.name) || ProjectSharingRoleType.EDIT,
+  );
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isOpenName, setIsOpenName] = React.useState(false);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  return (
+    <Tbody>
+      <Tr>
+        <Td dataLabel="Username">
+          {isEditing ? (
+            <Select
```

Comment:
Yes! that's right, and i agree, I should have kept a comment there, I'm sorry, I wanted to create the component, but as I wasn't sure if we were sticking with the typeahead or not I left it as it is, I was about to do this in case we kept some sort of typeahead